src/Targets/Microsoft.Managed.DesignTime.targets

```diff
@@ -25,6 +25,11 @@
     <SuppressOutOfDateMessageOnBuild Condition="'$(SuppressOutOfDateMessageOnBuild)' == ''">true</SuppressOutOfDateMessageOnBuild>
   </PropertyGroup>
 
+  <!-- If the project we're compiling has a ruleset, make sure we check it for up-to-date checks. -->
+  <ItemGroup Condition="'$(CodeAnalysisRuleSet)' != ''">
```

Comment:
It doesn't look like the old PS took rulesets into account for fast up to date, so this will add it there, too. I'll check to make sure it doesn't change anything, but I assume it's OK to extend the old PS as well?